docs/the-new-architecture/use-app-template.md

```diff
@@ -3,5 +3,61 @@ id: use-app-template
 title: Using the New App Template
 ---
 
-This section describes the new app template generated by the React Native CLI and how to use it properly.
-It should explain how to create a new app, which command should be issued to install the dependencies and to run it on the different platforms.
+This page will help you create a new React Native app that uses the new architecture.
+
+## Installing dependencies
+
+Start by setting up your development environment as described in [Getting Started](getting-started.md), under "React Native CLI Quickstart". Stop when you reach "React Native Command Line Interface", and resume here.
+
+<h3>React Native Command Line Interface</h3>
+
+React Native has a built-in command line interface. Rather than install and manage a specific version of the CLI globally, we recommend you access the current version at runtime using `npx`, which ships with Node.js. With `npx react-native <command>`, the current stable version of the CLI will be downloaded and executed at the time the command is run.
+
+<h2>Creating a new application</h2>
+
+> If you previously installed a global `react-native-cli` package, please remove it as it may cause unexpected issues.
+
+React Native has a built-in command line interface, which you can use to generate a new project. You can access it without installing anything globally using `npx`, which ships with Node.js. Let's create a new React Native project called "AwesomeProject":
+
+```shell
+npx react-native init AwesomeProject
+```
+
+## Configuring iOS
```

Comment:
For each of the platform-specific sections, I know we want to include information about the code the new template is generating. Is there any existing documentation/explanation of that? I don't know too much about it myself.

Reply:
From the configuration perspective, there is nothing to explain here. 

I would add a section specifying what the template creates after the configuration section. For what the template creates, it should basically create a copy of what is into this [folder](https://github.com/facebook/react-native/tree/main/template). Consider that the files that starts with an `_` will be transformed in hidden files (which starts with a `.` instead).

@cortinico Am I missing something important to mention about the template, here?

Reply:
Yeah, breaking this up into tabs works for me!

Reply:
Yeah, exactly. I think that all these `#ifdef` and all that code related to the [jsExecutorFactoryForBridge](http://localhost:3000/docs/next/new-architecture-app-modules-ios#2-install-turbomodulemanager-javascript-bindings) or to the [MyComponentRegistry](http://localhost:3000/docs/next/new-architecture-app-modules-ios#2-install-turbomodulemanager-javascript-bindings), for example, could be intimidating for a new react native user.

Reply:
Agreed. So would we rather skip explaining the native code in this document, then?

Reply:
> @cortinico Am I missing something important to mention about the template, here?

So I would maybe mention (in a box?) the upgrade helper:
https://react-native-community.github.io/upgrade-helper/?from=0.67.4&to=0.68.0
(This is essentially a diff of the new app template of 0.67 and 0.68).

Specifically we added banners there for files that are New Architecture only.

If you check through the diff, you'll see that there are several `.cpp`, and `.h` files inside the `/android` folder. Maybe it's worth mentioning that this new template already sets up all the C++ machinery needed to use the full power of the New Architecture.

If users want to know more about what each file is doing on Android, they're explained here:
https://reactnative.dev/docs/new-architecture-app-modules-android
https://reactnative.dev/docs/new-architecture-app-renderer-android

> we rather skip explaining the native code in this document

I'd say yes. We should mention that there is some native code, maybe mention why it's needed + link to those pages if users are interested on how to "recreate" this template step by step.

> I would add a section specifying what the template creates after the configuration section. For what the template creates, it should basically create a copy of what is into this [folder](https://github.com/facebook/react-native/tree/main/template?rgh-link-date=2022-04-06T10%3A12%3A46Z). Consider that the files that starts with an _ will be transformed in hidden files (which starts with a . instead).

Exactly. @cipolleschi nit: There are a couple of transformation happening as well (package name, app names, app ids, and so on). For the sake of this page, they can be ignored as they're minor implementation details.

Reply:
I added the relevant links @cortinico mentioned, as well as the corresponding options for iOS.

One thing I don't know yet that I think might be nice to explain is a brief statement on what is happening on each step. E.g., when I set `RCT_NEW_ARCH_ENABLED=1` on `pod install`, briefly, what does that change (e.g., "switches on Code Generation", or for Android, "`react-native run-android` now also compiles C++ code with Android NDK (what C++ code, I'm not sure how to explain?)). But not sure if that was intended only for other more detailed guides?

Reply:
I think that these are details slightly too deeply nested in how the New Architecture is implemented.
 
I'm not sure that it make sense to explain that the `RCT_NEW_ARCH_ENABLED` for iOS installs additional dependencies required by the New Architecture, generates some C++ code, and set some compilation flags so that the right code is compiled and executed. On Android it does more or less the same, but the compilation flags (which does not exists on that platform). 

But if you see the value there, let's add these short sentences. :+1:

Reply:
On the one hand, I think it can feel empowering for RN devs to have a high-level summary of what's being changed without having to go read the code.

On the other hand, you're right, it's a little down in the weeds, and if it were likely to change it's probably particularly not useful to know. And given that there's a manual migration guide, that should at least give anyone curious a better idea of the necessary changes.

Reply:
> And given that there's a manual migration guide, that should at least give anyone curious a better idea of the necessary changes.

☝️ You hit the point here. I think we don't need to deep dive too much on what is this extra flag doing, but just mention what it allows you to do. Users can still go to the migration guide to fully understand what is happenign here.